libraries/Microsoft.Bot.Connector/Authentication/CredentialProviders/TestCredentialProvider.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Connector.Authentication.CredentialProviders
+{
+    /// <summary>
+    /// A test implementation of the <see cref="ICredentialProvider"/> interface that doesn't implements IAppId interface.
+    /// </summary>
+    public class TestCredentialProvider : ICredentialProvider
```

Comment:
Please name this something like "TestCredentialProviderNoAppId" or similar. Make it painfully obvious...